### PR TITLE
Keyvault case sensitive names

### PR DESCRIPTION
**Pull request by mnriem**
## Summary
- Support for case sensitive key vault keys.

## Issue Type
- New Feature

## Starter Names
  - key vault spring boot starter

## Additional Information

### CODE REVIEW

**Comment by saragluna** [comment]
The integration tests failed.

**Comment by mnriem** [comment]
Not sure what to make of this one. It seems to try to get the property 'azure.cosmosdb.key' which is nowhere set in that test so it defaults to the value 'local' and thus the test fails. It has nothing to do with keyvault ...

**Comment by mnriem** [comment]
Missed the 'azure-cosmosdb-key' being set ... fixed a minor issue with boolean parsing to enable case sensitive key names. Committed that change, now waiting to see how that shakes out.

**Comment by saragluna** [comment]
We just merged PR #882 to maintain a cache of key vault secret names, which make this PR out of date with the master branch. Please help update this PR and resolve conflicts.

File: azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md [hunk 1]
```diff
@@ -96,6 +96,36 @@ azure.keyvault.allow.telemetry=false
 When telemetry is enabled, an HTTP request will be sent to URL `https://dc.services.visualstudio.com/v2/track`. So please make sure it's not blocked by your firewall.    
 Find more information about Azure Service Privacy Statement, please check [Microsoft Online Services Privacy Statement](https://www.microsoft.com/en-us/privacystatement/OnlineServices/Default.aspx). 
 
+<<<<<<< HEAD
```
Comment (saragluna) [inline azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md:99]: remove this line

File: azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md [hunk 2]
```diff
@@ -96,6 +96,36 @@ azure.keyvault.allow.telemetry=false
...
+An example of using a placeholder:
+
+```
+my.not.compliant.property=${myCompliantKeyVaultSecret}
+```
+
+The application will take care of getting the value that is backed by the 
+`myCompliantKeyVaultSecret` key name and assign its value to the non compliant
+`my.not.compliant.property`.
+=======
```
Comment (saragluna) [inline azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md:128]: here too

File: azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md [hunk 3]
```diff
@@ -122,3 +152,4 @@ Note if you decide to use multiple key vault support and you already have an
 existing configuration, please make sure you migrate that configuration to the
 multiple key vault variant. Mixing multiple key vaults with an existing single
 key vault configuration is a non supported scenario.
+>>>>>>> 194b454fe0b232315b21edd345a8948edc9a02b3
```
Comment (saragluna) [inline azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md:155]: and this one

**Comment by saragluna** [comment]
Is this PR ready to be merged?

**Comment by mnriem** [comment]
Yes it is

**Pull request merged**